### PR TITLE
Add fallback locator point recenter

### PR DESCRIPTION
**Pull request by jefff**
We normally use the center of the quads as the location of the tracking points, which is optimal for most cases and will account for a skew in the image. However, In some cases, a slight skew might not be real and instead be caused by image compression errors and/or low resolution. For those cases, we'd be better off centering the point exactly in the middle of the black area. This change  computes the location data for the naively centered points as it is little additional work and allows for multiple attempts at decoding harder images.

An example of this bug:
buggycenter

This change fixes 5 of the 47 currently failing tests, with no regressions.

### CODE REVIEW

File: src/locator/index.ts [hunk 1]
```diff
@@ -212,6 +212,20 @@ function scorePattern(point: Point, ratios: number[], matrix: BitMatrix) {
   }
 }
 
+function recenterLocation(matrix: BitMatrix, p: Point): Point {
```
Comment (cozmo) [inline src/locator/index.ts:215]: Any reason we wouldn't recenter vertically you?

File: src/locator/test.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ describe("locate", () => {
 
   it("locates a damaged QR code and guesses the finder pattern location", async () => {
```
Comment (cozmo) [inline src/locator/test.ts:66]: Don't feel too strongly but wouldn't hurt to add a test case for this - Would be as simple as taking one of the originally failing ones and committing it into the locator tests.

File: tests/end-to-end/report.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "counts": {
-    "failed": 47,
```
Comment (cozmo) [inline tests/end-to-end/report.json:3]: 😍

File: src/locator/test.ts (hunk 2 above)
Comment (jefff) [reply src/locator/test.ts:66]: a52cda5

File: src/locator/index.ts (hunk 1 above)
Comment (jefff) [reply src/locator/index.ts:215]: I didn't think it would matter, but after testing, it appears that also leads to a slight improvement.

a52cda5

**Pull request merged**